src/java/org/apache/cassandra/io/util/PathUtils.java

```diff
@@ -310,13 +312,39 @@ public static Throwable delete(Path file, Throwable accumulate, @Nullable RateLi
         return accumulate;
     }
 
+    private static void deleteRecursiveUsingNixCommand(Path path, boolean quietly)
```

Comment:
We always want to that fast don't we? 😄 

But this is a private method named after what it really does. And then we have a public remove method which decides which way to delete files should be chosen.

Reply:
Well, fair, nobody wants delete to be _slow_. But the reason this method exists is because on a specific platform it's significantly faster, and the only way a future maintainer would know that is an anthropological dig w/JIRA comments and descriptions to determine why something was changed. I'm fine leaving the name as is, but commenting why this seemingly redundant functionality was introduced would be nice.